### PR TITLE
Properly support automatic end correction when the typing causes multiple linked edits.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/4406

I have an integration test for this as well to verify the behavior.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by jasonmalinowski** [comment]
:+1:

File: src/EditorFeatures/Core/Shared/Utilities/LinkedEditsTracker.cs [hunk 1]
```diff
@@ -73,25 +73,31 @@ public bool TryGetTextChanged(TextContentChangedEventArgs args, out string repla
                 return false;
             }
 
+            // We want to find the single tracking span that encompasses all the edits made.  There 
+            // is no such tracking span (or there are multiple), then we consider the change invalid
+            // and we don't return any changed text.  If there is only one, then we find the text in
+            // the new document.
+            //
+            // Note there may be multiple intersection spans in the case where user typing causes
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Shared/Utilities/LinkedEditsTracker.cs]: Spelling: intersecting

File: src/EditorFeatures/Core/Shared/Utilities/LinkedEditsTracker.cs [hunk 2]
```diff
@@ -73,25 +73,31 @@ public bool TryGetTextChanged(TextContentChangedEventArgs args, out string repla
                 return false;
             }
 
+            // We want to find the single tracking span that encompasses all the edits made.  There
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Shared/Utilities/LinkedEditsTracker.cs]: *If* there is no such...

**Comment by Pilchie** [comment]
:+1:

**Comment by CyrusNajmabadi** [comment]
retest mac please

**Pull request merged**